Sources/StreamChat/Controllers/MessageController/MessageController.swift

```diff
@@ -306,6 +328,68 @@ public extension ChatMessageController {
             self.callback { completion?(result.error) }
         }
     }
+
+    /// Loads the next page of reactions starting from the current fetched reactions.
+    ///
+    /// - Parameters:
+    ///   - limit: The reactions page size.
+    ///   - completion: The completion is called when the network request is finished.
+    ///   If the request fails, the completion will be called with an error, if it succeeds it is
+    ///   called without an error and the delegate is notified of reactions changes.
+    func loadNextReactions(
+        limit: Int = 25,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        if hasLoadedAllReactions {
+            callback { completion?(nil) }
+            return
+        }
+
+        loadReactions(limit: limit, offset: reactions.count) { result in
+            switch result {
+            case let .success(reactions):
+                let currentReactions = Set(self.reactions)
+                for reaction in reactions where !currentReactions.contains(reaction) {
+                    self.reactions.append(reaction)
+                }
```

Comment:
Hm, if we append new reactions like that the delegate will fire multiple times, right? 🤔 Is it on purpose?

Reply:
Good catch 🤔 Let me check this

Reply:
Fixed ✅ Thank you!